### ISSUE TITLE
Requirements for the images / error: vector subscript out of range

### ISSUE THREAD

**Issue by maxklock** [chaowang15/RGBDPlaneDetection#5]
What are the requirements for the images? I am currently trying with my own images, but I get an error message: vector subscript out of range
The error comes from _plane_detection.cpp_ line 226. `new_pidx` gets -1.

My images are 1280x720 and I have adjusted the corresponding constants (_plane_detection.h_ lines 25-30). But even if I scale my images to 640x480 before, I get the error.

In addition the plane image looks very strange.
1280x720 without adjusted constants: (no change, cloned 25.05.2020)
![f_00000-plane](https://user-images.githubusercontent.com/7167301/82817698-bc094680-9e9d-11ea-8dad-36f1f84b3d94.png)

1280x720 with adjusted constants (similar to the image when scaled to 640x480):
![f_00000-plane](https://user-images.githubusercontent.com/7167301/82817772-d93e1500-9e9d-11ea-9d31-8fdc77c1c686.png)

The only change I made to adjust constants:
>diff --git a/RGBDPlaneDetection/plane_detection.h b/RGBDPlaneDetection/plane_detection.h
>index 8e1d876..b58ebf8 100644
>--- a/RGBDPlaneDetection/plane_detection.h
>+++ b/RGBDPlaneDetection/plane_detection.h
>@@ -22,12 +22,12 @@ const float kInfVal = 1000000; // an infinite large value used in MRF optimizati
>
> // Camera intrinsic parameters.
> // All BundleFusion data uses the following parameters.
>-const double kFx = 583;
>-const double kFy = 583;
>-const double kCx = 320;
>-const double kCy = 240;
>-const int kDepthWidth = 640;
>-const int kDepthHeight = 480;
>+const double kFx = 633.085;
>+const double kFy = 633.085;
>+const double kCx = 640.891;
>+const double kCy = 356.606;
>+const int kDepthWidth = 1280;
>+const int kDepthHeight = 720;
>
> #if defined(__linux__) || defined(__APPLE__)
> #define _isnan(x) isnan(x)

At last the original pictures:
![f_00000_color](https://user-images.githubusercontent.com/7167301/82818300-cd9f1e00-9e9e-11ea-910a-70f09f373d5e.png)
![f_00000_depth16](https://user-images.githubusercontent.com/7167301/82818307-ced04b00-9e9e-11ea-9346-6f4db5c4e968.png)
(Recorded in an simulation, based on the [Stanford data set](http://buildingparser.stanford.edu/dataset.html))

**Comment by chaowang15** [comment]
@maxklock Apologize to reply you late. I just tested your image in my place and both methos with or without optimization work well. Here are my results: 

No optimization:
![te-plane](https://user-images.githubusercontent.com/5984316/83083946-5ef7c700-a03c-11ea-8aaa-f9fbf07c5395.png)

With Optimization:
![te-plane-opt](https://user-images.githubusercontent.com/5984316/83083953-63bc7b00-a03c-11ea-9294-5614756dcea1.png)

The only place I changed in the code is the intrinsics. I scaled your color image to 640x360 with same size of depth image.
